nav2_gps_waypoint_follower/src/gps_waypoint_follower.cpp

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) 2020 Fetullah Atas
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include <string>
+#include <memory>
+#include "nav2_gps_waypoint_follower/gps_waypoint_follower.hpp"
+
+namespace nav2_gps_waypoint_follower
+{
+GPSWaypointFollower::GPSWaypointFollower()
+: nav2_util::LifecycleNode("GPSWaypointFollower", "", false)
+{
+  RCLCPP_INFO(get_logger(), "Creating");
+}
+
+GPSWaypointFollower::~GPSWaypointFollower()
+{
+  RCLCPP_INFO(get_logger(), "Destroying");
+}
+
+nav2_util::CallbackReturn
+GPSWaypointFollower::on_configure(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(get_logger(), "Configuring");
+  std::vector<std::string> new_args = rclcpp::NodeOptions().arguments();
+  new_args.push_back("--ros-args");
+  new_args.push_back("-r");
+  new_args.push_back(std::string("__node:=") + this->get_name() + "_rclcpp_node");
+  new_args.push_back("--");
+  client_node_ = std::make_shared<rclcpp::Node>(
+    "_", "", rclcpp::NodeOptions().arguments(new_args));
+
+  waypoint_follower_goal_ = ClientT::Goal();
+  waypoint_follower_action_client_ = rclcpp_action::create_client<ClientT>(
+    client_node_, "FollowWaypoints");
+
+  callback_group_ = this->create_callback_group(
```

Comment:
You're using a client node already

Reply:
I encountered some error if not using callback group, since I am being client to two different services(`robot_localization` and `FollowWaypoints`), calling `spin_until_future_complete` twiche on client_node let me to run into some issue; 
`Node has already been added to an executor.`
This parts are likely to change, please see my comment